### PR TITLE
OP-1043 | Remove deleted group from user edit group jcombo box

### PR DESCRIPTION
**Pull request by SteveGT96**
See [OP-1043](https://openhospital.atlassian.net/browse/OP-1043)

[OP-1043]: https://openhospital.atlassian.net/browse/OP-1043?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/java/org/isf/menu/gui/UserEdit.java [hunk 1]
```diff
@@ -340,20 +340,24 @@ private JComboBox<UserGroup> getUserGroupComboBox() {
...
 			try {
-				List<UserGroup> group = userBrowsingManager.getUserGroup();
+				List<UserGroup> groups = userBrowsingManager.getUserGroup();
 				if (insert) {
-					if (group != null) {
-						for (UserGroup elem : group) {
-							userGroupComboBox.addItem(elem);
+					if (groups != null) {
+						for (UserGroup group : groups) {
+							if(!group.isDeleted()) {
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/menu/gui/UserEdit.java]:
```suggestion
							if (!group.isDeleted()) {
```

File: src/main/java/org/isf/menu/gui/UserEdit.java [hunk 2]
```diff
@@ -340,20 +340,24 @@ private JComboBox<UserGroup> getUserGroupComboBox() {
...
 				} else {
 					UserGroup selectedUserGroup = null;
-					if (group != null) {
-						for (UserGroup elem : group) {
-							userGroupComboBox.addItem(elem);
-							if (user.getUserGroupName().equals(elem)) {
-								selectedUserGroup = elem;
+					if (groups != null) {
+						for (UserGroup group : groups) {
+							if(!group.isDeleted()) {
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/menu/gui/UserEdit.java]:
```suggestion
							if (!group.isDeleted()) {
```

File: src/main/java/org/isf/menu/gui/UserEdit.java [hunk 3]
```diff
@@ -340,20 +340,24 @@ private JComboBox<UserGroup> getUserGroupComboBox() {
...
+								userGroupComboBox.addItem(group);
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/menu/gui/UserEdit.java:357]: Isn't this else block used if `insert` is false?   Then why add the group to the combobox?  What am I mssing?

Comment (SteveGT96) [reply src/main/java/org/isf/menu/gui/UserEdit.java:357]: I'm not understanding what you mean, sorry. I've just wrapped the instruction block with the 'if' statement.

Comment (dbmalkovsky) [reply src/main/java/org/isf/menu/gui/UserEdit.java:357]: nevermind.......

**Review by mwithi — APPROVED** [review]
LGTM. Thanks a lot!

**Pull request merged**